### DISCUSSION TITLE
Ability to add custom tags (variables)

### DISCUSSION THREAD

**Discussion by davidlamkin** [music-assistant/support#1559] [Feature requests and ideas]
It would be great if I could add tags/variables to Artists/Albums/Tracks - or even to music providers themselves. These "tags" could be a part of the Hass media_player attributes.

My music library contains music for my whole household and we all have differences in taste. I am trying to scrobble tracks, but whenever someone else plays somethings, I have to delete it in Last.fm. If I could scrobble based on _who_ is playing what, that would be awesome.

**Comment by OzGav** [comment]
You can add custom tags via a tag editor such as MP3Tag. How do you envisage this would be used though? How is last.fm currently receiving the data that it uses to scrobble?

**Reply by davidlamkin** [reply]
My apologies. I meant being able to set a variable that can carry over to an HA attribute. One use for this functionality could be to define variables like separating libraries for different users. Another could be to define actions for certain artists for automation (like scrobbling). Etc...

**Reply by OzGav** [reply]
How do you envisage that this variable gets its value? There has been suggestions before that HA feeds the user that is logged in to MA as there is no login functionality currently for MA.

**Reply by davidlamkin** [reply]
I would see it like a tagging system within Mass. Then in HA you could have the list of Mass tags in the attributes.

**Reply by OzGav** [reply]
OK there is no such functionality on the roadmap at the moment. Will have to wait and see if this idea gets any traction from other users,

**Comment by juergenpabel** [comment]
I have a specific use-case for this: I am developing a system that uses RFID/NFC cards that can be used to trigger a certain media item to be played. My current implementation is not released yet and was essentially and all-in-one implementation. After learning about MA, I would like to ditch my own implementation parts for music providers and players and just use MA instead (so far I only implemented local mp3s and spotify for "providers" and only sonos for "players"). This leaves essentially just the part that acts upon a RFID/NFC card being inserted/removed from a reader - which would be great because that's the core "value" of my project. However, I still need to maintain associations between cards and the media piece to be played...and that's what I would now like to manage with MA - so that's my use case for a custom tag on music items in MA. Technically, I would store the card's UID (supposedly unique for any card in the world...more likely for any card in anyone's possession) on the music item in MA. I could also store the MA music items data on the card itself, but that requires writing to the card and that doesn't work reliably (with all card types that are out there - just reading the card UID works for any card type).

And by the way (if you were wondering): Each card reader is associated with a music player (sonos in my case) so the question of where should that be played is part of the readers configuration. Essentially, my card readers are attached to microcontrollers that than communicate the events (card inserted/removed) to a service running on a mini-server (like a raspi), which than initiates the corresponding media action (and this would be than re-implemented to be a MA client).

So, I would second this feature request - and I'd be fine with an implementation that isn't exposed in the web UI (just the capability to assign a custom property - key/value style - to any music item in HA via the API). I can also offer to implement this myself and submit a PR, but being unfamiliar with MA's codebase, I would than greatly appreciate some pointers as to how it should be implemented "best".

**Reply by juergenpabel** [reply]
<img width="191" height="340" alt="reader" src="https://github.com/user-attachments/assets/dadef745-57af-4ff4-8378-a8c056e924f9" />

This is one of my readers (it's in regular use by my kids) - the NFC/RFID reader plus microcontroller (with WLAN) are contained in the 3d-printed enclosure and all it needs (physically) is a USB cable for power. The cards are inserted from the top, so they are also than in the "corrected" position for the power/communications field.

**Reply by OzGav** [reply]
Please upvote the OP as that guides the development
